app/services/courses_query_logger.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class CoursesQueryLogger
```

Comment:
I created this class because of my own necessity.

To see which filters is applied, inspect better in the log the whole query and also to explain to understand the query performance.

This is already helping me to see if the code applied the filter or not, see the query in the midst of the log chaos and also explain and understand a course problem with the courses <-> course_subjects <-> subjects (because of the after_initialize block in the course model).

Reply:
This class is coupled to the `CoursesQuery`, yet it is distinctly detached. Does it make sense to make this a subclass of `CoursesQuery` itself? `CoursesQuery::Logger` possibly?

Reply:
Additionally, this does feel like it is a bit preemptive.

Reply:
> You might want to show a screenshot of what the log output looks like in the PR description.

Done.

> Additionally, this does feel like it is a bit preemptive.

Maybe, although it already proves its usefulness so I plan to maintain. The advantage is this can be easily removed if don't prove to be useful anymore.

> This class is coupled to the CoursesQuery, yet it is distinctly detached. Does it make sense to make this a subclass of CoursesQuery itself? CoursesQuery::Logger possibly?

Done!